src/GitHub.App/Models/RepositoryHosts.cs

```diff
@@ -157,11 +157,15 @@ public IRepositoryHost LookupHost([AllowNull] HostAddress address)
                     );
                     if (successful)
                     {
+                        // We need to add the connection before assigning to GitHubHost or
+                        // EnterpriseHost so that anything listening to those will find the
+                        // associated connection.
+                        connectionManager.AddConnection(address, usernameOrEmail);
```

Comment:
It's more of a "if it can race, I'm sure we'll make code that races" thing. It's the law!